src/core/server/status/plugins_status.ts

```diff
@@ -330,12 +330,12 @@ export class PluginsStatusService {
    * @param {ServiceStatus} reportedStatus The newly reported status for that plugin
    */
   private updatePluginReportedStatus(plugin: PluginName, reportedStatus: ServiceStatus): void {
-    const previousReportedLevel = this.pluginData[plugin].reportedStatus?.level;
+    const previousReportedStatus = this.pluginData[plugin].reportedStatus;
 
     this.pluginData[plugin].reportedStatus = reportedStatus;
     this.pluginStatus[plugin] = reportedStatus;
 
-    if (reportedStatus.level !== previousReportedLevel) {
+    if (!isEqual(previousReportedStatus, reportedStatus)) {
```

Comment:
Note that, for a given plugin *P*, the `updatePluginsStatuses` recalculates the statuses of the subtree of dependencies of P.

The question here would be: 
Are the rest of plugins' statuses affected by the changes in P's `summary` field?
If they are only affected by the `level`, then what we should be doing is:
* Not updating the tree of dependencies.
* Emitting the new "state" to the output Observables, so that the new `summary` is propagated.

If that's what we want, I'd suggest we remove the "emit" responsibility from the `updatePluginsStatuses` and pass it on to the callers of that function, aka:
* `updateCoreAndPluginStatuses` (or even call in in the subscribe handler defined in the constructor)
* `updatePluginReportedStatus` (or maybe even better, in the subscribe handler defined in the `set` method).

Reply:
@nchaulet I can take care of performing the aforementioned changes, if you are okay with that.

Reply:
> @nchaulet I can take care of performing the aforementioned changes, if you are okay with that.

Yes of course, go ahead :)